src/common/transport/h5_transport.cpp

```diff
@@ -652,6 +652,11 @@ void H5Transport::startStateMachine()
     {
         stateMachineThread = new std::thread(std::bind(&H5Transport::stateMachineWorker, this));
     }
+    else
+    {
+        // Terminate if the state machine already exists, this should not happen.
+        std::terminate();
```

Comment:
Add a log line here to give the user a hint on why the application is terminated?

Reply:
✔️